qa/tasks/cephfs/test_nfs.py

```diff
@@ -627,6 +666,18 @@ def test_data_read_write(self):
         self._test_data_read_write(self.pseudo_path, port, ip)
         self._test_delete_cluster()
 
+    def test_async_io_fio(self):
+        '''
+        Test async io using fio. Expect completion without hang or crash
+        '''
+        self._test_create_cluster()
+        self._create_export(export_id='1', create_fs=True,
+                            extra_cmd=['--pseudo-path', self.pseudo_path])
+        port, ip = self._get_port_ip_info()
+        self._check_nfs_cluster_status('running', 'NFS Ganesha cluster restart failed')
+        self._test_fio(self.pseudo_path, port, ip)
```

Comment:
@kotreshhr - I see `fio` failing with

```
2024-11-18T09:40:48.299 INFO:teuthology.orchestra.run.smithi072.stderr:fio: io_u error on file /mnt/fio/fio.randrw.test: Input/output error: read offset=933888, buflen=16384
2024-11-18T09:40:48.299 INFO:teuthology.orchestra.run.smithi072.stderr:fio: io_u error on file /mnt/fio/fio.randrw.test: Input/output error: read offset=950272, buflen=16384
2024-11-18T09:40:48.299 INFO:teuthology.orchestra.run.smithi072.stderr:fio: io_u error on file /mnt/fio/fio.randrw.test: Input/output error: read offset=966656, buflen=16384
2024-11-18T09:40:48.299 INFO:teuthology.orchestra.run.smithi072.stderr:fio: io_u error on file /mnt/fio/fio.randrw.test: Input/output error: read offset=983040, buflen=16384
2024-11-18T09:40:48.299 INFO:teuthology.orchestra.run.smithi072.stderr:fio: io_u error on file /mnt/fio/fio.randrw.test: Input/output error: read offset=999424, buflen=16384
2024-11-18T09:40:48.299 INFO:teuthology.orchestra.run.smithi072.stderr:fio: io_u error on file /mnt/fio/fio.randrw.test: Input/output error: read offset=1015808, buflen=16384
2024-11-18T09:40:48.300 INFO:teuthology.orchestra.run.smithi072.stderr:fio: io_u error on file /mnt/fio/fio.randrw.test: Input/output error: read offset=1032192, buflen=16384
2024-11-18T09:40:48.300 INFO:teuthology.orchestra.run.smithi072.stderr:fio: io_u error on file /mnt/fio/fio.randrw.test: Input/output error: read offset=1048576, buflen=16384
2024-11-18T09:40:48.300 INFO:teuthology.orchestra.run.smithi072.stdout:fio: pid=87100, err=5/file:io_u.c:1845, func=io_u error, error=Input/output error
2024-11-18T09:40:48.351 INFO:teuthology.orchestra.run.smithi072.stdout:
2024-11-18T09:40:48.351 INFO:teuthology.orchestra.run.smithi072.stdout:job: (groupid=0, jobs=1): err= 5 (file:io_u.c:1845, func=io_u error, error=Input/output error): pid=87100: Mon Nov 18 09:40:48 2024
2024-11-18T09:40:48.351 INFO:teuthology.orchestra.run.smithi072.stdout:  write: IOPS=381, BW=6109KiB/s (6256kB/s)(1008KiB/165msec); 0 zone resets
2024-11-18T09:40:48.351 INFO:teuthology.orchestra.run.smithi072.stdout:  cpu          : usr=0.00%, sys=2.44%, ctx=3, majf=0, minf=1
2024-11-18T09:40:48.351 INFO:teuthology.orchestra.run.smithi072.stdout:  IO depths    : 1=0.8%, 2=1.6%, 4=3.1%, 8=6.2%, 16=12.5%, 32=25.0%, >=64=50.8%
2024-11-18T09:40:48.351 INFO:teuthology.orchestra.run.smithi072.stdout:     submit    : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=0.0%
2024-11-18T09:40:48.352 INFO:teuthology.orchestra.run.smithi072.stdout:     complete  : 0=33.3%, 4=33.3%, 8=0.0%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=33.3%
2024-11-18T09:40:48.352 INFO:teuthology.orchestra.run.smithi072.stdout:     issued rwts: total=65,63,0,0 short=0,0,0,0 dropped=0,0,0,0
2024-11-18T09:40:48.352 INFO:teuthology.orchestra.run.smithi072.stdout:     latency   : target=0, window=0, percentile=100.00%, depth=128
2024-11-18T09:40:48.352 INFO:teuthology.orchestra.run.smithi072.stdout:
2024-11-18T09:40:48.352 INFO:teuthology.orchestra.run.smithi072.stdout:Run status group 0 (all jobs):
2024-11-18T09:40:48.352 INFO:teuthology.orchestra.run.smithi072.stdout:  WRITE: bw=6109KiB/s (6256kB/s), 6109KiB/s-6109KiB/s (6256kB/s-6256kB/s), io=1008KiB (1032kB), run=165-165msec
2024-11-18T09:40:48.387 DEBUG:teuthology.orchestra.run:got remote process result: 1
2024-11-18T09:40:48.388 DEBUG:teuthology.orchestra.run.smithi072:> sudo rm -rf /mnt/fio
2024-11-18T09:40:48.423 DEBUG:teuthology.orchestra.run.smithi072:> sudo umount /mnt
```

/a/vshankar-2024-11-18_08:59:55-fs-wip-vshankar-testing-20241118.055430-debug-testing-default-smithi/7997433

Reply:
ok, I will take a look.

Reply:
@vshankar 

I see that the connection reset errors in mds and osd

mds
```
[khiremat@vossi04 142b0ad6-a58f-11ef-bb9f-d5e06f7e0c9a]$ zgrep -i error ceph-mds.nfs-cephfs.smithi072.ckwfht.log.gz 
2024-11-18T09:41:08.348+0000 7fbe0d200640  1 -- [v2:172.21.15.72:6832/1087774968,v1:172.21.15.72:6833/1087774968] >> [v2:172.21.15.72:6828/2931952847,v1:172.21.15.72:6829/2931952847] conn(0x5619f0b56800 msgr2=0x5619f591c000 unknown :-1 s=STATE_CONNECTION_ESTABLISHED l=1).read_bulk reading from fd=24 : -104 Unknown error -104
2024-11-18T09:41:15.056+0000 7fbe0dc00640  1 -- [v2:172.21.15.72:6832/1087774968,v1:172.21.15.72:6833/1087774968] >> [v2:172.21.15.72:6802/1506627565,v1:172.21.15.72:6803/1506627565] conn(0x5619f594e800 msgr2=0x5619f591c580 unknown :-1 s=STATE_CONNECTION_ESTABLISHED l=1).read_bulk reading from fd=24 : -104 Unknown error -104
2024-11-18T09:41:26.236+0000 7fbe0e600640  1 -- [v2:172.21.15.72:6832/1087774968,v1:172.21.15.72:6833/1087774968] >> [v2:172.21.15.72:6810/3302151231,v1:172.21.15.72:6811/3302151231] conn(0x5619f5dbf000 msgr2=0x5619f5b0ec00 unknown :-1 s=STATE_CONNECTION_ESTABLISHED l=1).read_bulk reading from fd=24 : -104 Unknown error -104


[khiremat@vossi04 142b0ad6-a58f-11ef-bb9f-d5e06f7e0c9a]$ zgrep -i error ceph-mds.nfs-cephfs.smithi072.tmgpun.log.gz 
2024-11-18T09:41:08.348+0000 7fe570600640  1 -- [v2:172.21.15.72:6830/2522609555,v1:172.21.15.72:6831/2522609555] >> [v2:172.21.15.72:6828/2931952847,v1:172.21.15.72:6829/2931952847] conn(0x562b57478800 msgr2=0x562b5c23e000 unknown :-1 s=STATE_CONNECTION_ESTABLISHED l=1).read_bulk reading from fd=24 : -104 Unknown error -104
```

In osd, along with connection reset (-104), I see pg_update_log_missing messages and find_object_context got error -2 in osd1

osd:
```
osd0
2024-11-18T09:39:30.959+0000 7fecc6a00640  1 -- [v2:172.21.15.72:6802/1506627565,v1:172.21.15.72:6803/1506627565] >> [v2:172.21.15.72:6828/2537027960,v1:172.21.15.72:6829/2537027960] conn(0x5567002eb800 msgr2=0x55670036e100 unknown :-1 s=STATE_CONNECTION_ESTABLISHED l=1).read_bulk reading from fd=63 : -104 Unknown error -104
2024-11-18T09:39:51.858+0000 7fecc6a00640  1 -- [v2:172.21.15.72:6804/1506627565,v1:172.21.15.72:6805/1506627565] <== osd.1 v2:172.21.15.72:6812/3302151231 105 ==== pg_update_log_missing(1.0 epoch 23/19 rep_tid 59 entries 23'60 (0'0) error    1:30306672:devicehealth::main.db.0000000000000000:head by client.14362.0:1 0.000000 -17 [r=-17+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false trim_to 0'0 pg_committed_to 22'59) ==== 341+0+0 (crc 0 0 0) 0x55670030f400 con 0x556700938400




osd1

2024-11-18T09:39:30.959+0000 7f835da00640  1 -- [v2:172.21.15.72:6810/3302151231,v1:172.21.15.72:6811/3302151231] >> [v2:172.21.15.72:6828/2537027960,v1:172.21.15.72:6829/2537027960] conn(0x55f64b88b400 msgr2=0x55f64b87a100 unknown :-1 s=STATE_CONNECTION_ESTABLISHED l=1).read_bulk reading from fd=66 : -104 Unknown error -104
2024-11-18T09:39:51.858+0000 7f8340400640 10 osd.1 pg_epoch: 23 pg[1.0( v 22'59 (0'0,22'59] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 crt=22'59 lcod 22'58 mlcod 22'58 active+clean] do_osd_ops error: (17) File exists
2024-11-18T09:39:51.858+0000 7f8340400640 20 osd.1 pg_epoch: 23 pg[1.0( v 22'59 (0'0,22'59] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 crt=22'59 lcod 22'58 mlcod 22'58 active+clean] record_write_error r=-17
2024-11-18T09:39:51.858+0000 7f8340400640 20 osd.1 pg_epoch: 23 pg[1.0( v 22'59 (0'0,22'59] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 crt=22'59 lcod 22'58 mlcod 22'58 active+clean] record_write_error op_returns=[r=-17+0b,r=0+0b,r=0+0b,r=0+0b,r=0+0b,r=0+0b,r=0+0b,r=0+0b]
2024-11-18T09:39:51.858+0000 7f8340400640 10 osd.1 pg_epoch: 23 pg[1.0( v 22'59 (0'0,22'59] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 crt=22'59 lcod 22'58 mlcod 22'58 active+clean] submit_log_entries 23'60 (0'0) error    1:30306672:devicehealth::main.db.0000000000000000:head by client.14362.0:1 0.000000 -17 [r=-17+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false
2024-11-18T09:39:51.858+0000 7f8340400640 10 osd.1 pg_epoch: 23 pg[1.0( v 22'59 (0'0,22'59] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 crt=22'59 lcod 22'58 mlcod 22'58 active+clean] PeeringState::merge_new_log_entries 23'60 (0'0) error    1:30306672:devicehealth::main.db.0000000000000000:head by client.14362.0:1 0.000000 -17 [r=-17+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false
2024-11-18T09:39:51.858+0000 7f8340400640 20 update missing, append 23'60 (0'0) error    1:30306672:devicehealth::main.db.0000000000000000:head by client.14362.0:1 0.000000 -17 [r=-17+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false
2024-11-18T09:39:51.858+0000 7f8340400640  1 -- [v2:172.21.15.72:6812/3302151231,v1:172.21.15.72:6813/3302151231] --> [v2:172.21.15.72:6804/1506627565,v1:172.21.15.72:6805/1506627565] -- pg_update_log_missing(1.0 epoch 23/19 rep_tid 59 entries 23'60 (0'0) error    1:30306672:devicehealth::main.db.0000000000000000:head by client.14362.0:1 0.000000 -17 [r=-17+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false trim_to 0'0 pg_committed_to 22'59) -- 0x55f645c66000 con 0x55f64b1ba400
2024-11-18T09:39:51.858+0000 7f8340400640  1 -- [v2:172.21.15.72:6812/3302151231,v1:172.21.15.72:6813/3302151231] --> [v2:172.21.15.72:6820/268304520,v1:172.21.15.72:6821/268304520] -- pg_update_log_missing(1.0 epoch 23/19 rep_tid 59 entries 23'60 (0'0) error    1:30306672:devicehealth::main.db.0000000000000000:head by client.14362.0:1 0.000000 -17 [r=-17+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false trim_to 0'0 pg_committed_to 22'59) -- 0x55f64bad9800 con 0x55f64b6cd400
2024-11-18T09:39:51.858+0000 7f833b400640 20 osd.1 pg_epoch: 23 pg[1.0( v 23'61 (0'0,23'61] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 pct=23'60 lua=23'60 crt=23'61 lcod 22'59 mlcod 22'59 active+clean] rollforward: entry=23'60 (0'0) error    1:30306672:devicehealth::main.db.0000000000000000:head by client.14362.0:1 0.000000 -17 [r=-17+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b, r=0+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false
2024-11-18T09:39:51.862+0000 7f833b400640 20 osd.1 pg_epoch: 23 pg[1.0( v 23'61 (0'0,23'61] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 crt=23'61 lcod 23'60 mlcod 23'60 active+clean] do_op: find_object_context got error -2
2024-11-18T09:39:51.870+0000 7f833b400640 20 osd.1 pg_epoch: 23 pg[1.0( v 23'63 (0'0,23'63] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 crt=23'63 lcod 23'62 mlcod 23'62 active+clean] do_op: find_object_context got error -2
2024-11-18T09:39:51.886+0000 7f8340400640 20 osd.1 pg_epoch: 23 pg[1.0( v 23'65 (0'0,23'65] local-lis/les=19/20 n=2 ec=19/19 lis/c=19/19 les/c/f=20/20/0 sis=19) [1,0,2] r=0 lpr=19 crt=23'65 lcod 23'64 mlcod 23'64 active+clean] do_op: find_object_context got error -2
2024-11-18T09:40:04.110+0000 7f8339600640 20 osd.1 pg_epoch: 26 pg[2.1b( empty local-lis/les=24/25 n=0 ec=24/24 lis/c=24/24 les/c/f=25/25/0 sis=24) [1,0,2] r=0 lpr=24 crt=0'0 mlcod 0'0 active+clean] do_op: find_object_context got error -2
2024-11-18T09:40:07.114+0000 7f833e600640 10 osd.1 pg_epoch: 26 pg[2.1b( empty local-lis/les=24/25 n=0 ec=24/24 lis/c=24/24 les/c/f=25/25/0 sis=24) [1,0,2] r=0 lpr=24 crt=0'0 mlcod 0'0 active+clean] do_osd_ops error: (2) No such file or directory
2024-11-18T09:40:07.114+0000 7f833e600640 20 osd.1 pg_epoch: 26 pg[2.1b( empty local-lis/les=24/25 n=0 ec=24/24 lis/c=24/24 les/c/f=25/25/0 sis=24) [1,0,2] r=0 lpr=24 crt=0'0 mlcod 0'0 active+clean] record_write_error r=-2
2024-11-18T09:40:07.114+0000 7f833e600640 20 osd.1 pg_epoch: 26 pg[2.1b( empty local-lis/les=24/25 n=0 ec=24/24 lis/c=24/24 les/c/f=25/25/0 sis=24) [1,0,2] r=0 lpr=24 crt=0'0 mlcod 0'0 active+clean] record_write_error op_returns=[r=-2+0b]
2024-11-18T09:40:07.114+0000 7f833e600640 10 osd.1 pg_epoch: 26 pg[2.1b( empty local-lis/les=24/25 n=0 ec=24/24 lis/c=24/24 les/c/f=25/25/0 sis=24) [1,0,2] r=0 lpr=24 crt=0'0 mlcod 0'0 active+clean] submit_log_entries 26'1 (0'0) error    2:d9d43a17:test::rec-0000000000000001%3anfs.test.0:head by client.14387.0:20 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false
2024-11-18T09:40:07.114+0000 7f833e600640 10 osd.1 pg_epoch: 26 pg[2.1b( empty local-lis/les=24/25 n=0 ec=24/24 lis/c=24/24 les/c/f=25/25/0 sis=24) [1,0,2] r=0 lpr=24 crt=0'0 mlcod 0'0 active+clean] PeeringState::merge_new_log_entries 26'1 (0'0) error    2:d9d43a17:test::rec-0000000000000001%3anfs.test.0:head by client.14387.0:20 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false
2024-11-18T09:40:07.114+0000 7f833e600640 20 update missing, append 26'1 (0'0) error    2:d9d43a17:test::rec-0000000000000001%3anfs.test.0:head by client.14387.0:20 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false



osd2:

2024-11-18T09:40:50.992+0000 7f068c200640 20 osd.2 op_wq(0) _process OpSchedulerItem(3.10 PGOpItem(op=pg_update_log_missing(3.10 epoch 32/27 rep_tid 1356 entries 32'9 (0'0) error    3:0cf6ad0f:::10000000000.00000000:head by mds.0.9:159 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false trim_to 0'0 pg_committed_to 29'8) v3) class_id 2 prio 127 cost 0 e
32) pg 0x563e61c94000
2024-11-18T09:40:50.992+0000 7f068c200640 10 osd.2 32 dequeue_op pg_update_log_missing(3.10 epoch 32/27 rep_tid 1356 entries 32'9 (0'0) error    3:0cf6ad0f:::10000000000.00000000:head by mds.0.9:159 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false trim_to 0'0 pg_committed_to 29'8) v3 prio 127 cost 0 latency 0.000199 pg_update_log_missing(3.10 epoch 
32/27 rep_tid 1356 entries 32'9 (0'0) error    3:0cf6ad0f:::10000000000.00000000:head by mds.0.9:159 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false trim_to 0'0 pg_committed_to 29'8) v3 pg pg[3.10( v 29'8 (0'0,29'8] local-lis/les=27/28 n=4 ec=27/27 lis/c=27/27 les/c/f=28/28/0 sis=27) [1,0,2] r=2 lpr=27 pct=0'0 lua=0'0 crt=29'8 lcod 29'7 active mbc=
{}]
2024-11-18T09:40:50.992+0000 7f068c200640 10 osd.2 pg_epoch: 32 pg[3.10( v 29'8 (0'0,29'8] local-lis/les=27/28 n=4 ec=27/27 lis/c=27/27 les/c/f=28/28/0 sis=27) [1,0,2] r=2 lpr=27 pct=0'0 lua=0'0 crt=29'8 lcod 29'7 active mbc={}] _handle_message: pg_update_log_missing(3.10 epoch 32/27 rep_tid 1356 entries 32'9 (0'0) error    3:0cf6ad0f:::10000000000.00000000:head by mds.0.9:159 0.000000 -2 [r=-2+0b] ObjectCleanRegions cl
ean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false trim_to 0'0 pg_committed_to 29'8) v3
2024-11-18T09:40:50.992+0000 7f068c200640 20 update missing, append 32'9 (0'0) error    3:0cf6ad0f:::10000000000.00000000:head by mds.0.9:159 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false
2024-11-18T09:40:50.992+0000 7f068c200640 10 osd.2 32 dequeue_op pg_update_log_missing(3.10 epoch 32/27 rep_tid 1356 entries 32'9 (0'0) error    3:0cf6ad0f:::10000000000.00000000:head by mds.0.9:159 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_omap: true, new_object: false trim_to 0'0 pg_committed_to 29'8) v3 finish
2024-11-18T09:40:52.552+0000 7f068ea00640 20 osd.2 pg_epoch: 32 pg[3.14( v 32'110 (0'0,32'110] local-lis/les=27/28 n=4 ec=27/27 lis/c=27/27 les/c/f=28/28/0 sis=27) [1,2,0] r=1 lpr=27 pct=32'107 lua=32'109 crt=32'110 lcod 32'109 active mbc={}] rollforward: entry=32'109 (0'0) error    3:2e992e0f:::500.00000003:head by mds.0.9:152 0.000000 -2 [r=-2+0b] ObjectCleanRegions clean_offsets: [(0, 18446744073709551615)], clean_om
ap: true, new_object: false
2024-11-18T09:41:15.056+0000 7f06afc00640  1 -- 172.21.15.72:0/268304520 >> [v2:172.21.15.72:6808/1506627565,v1:172.21.15.72:6809/1506627565] conn(0x563e7081dc00 msgr2=0x563e61f84b00 unknown :-1 s=STATE_CONNECTION_ESTABLISHED l=1).read_bulk reading from fd=58 : -104 Unknown error -104
2024-11-18T09:41:15.056+0000 7f06af200640  1 -- 172.21.15.72:0/268304520 >> [v2:172.21.15.72:6806/1506627565,v1:172.21.15.72:6807/1506627565] conn(0x563e62add000 msgr2=0x563e61f85b80 unknown :-1 s=STATE_CONNECTION_ESTABLISHED l=1).read_bulk reading from fd=63 : -104 Unknown error -104
```

I retried the same test and it has successfully passed:

https://pulpito.ceph.com/khiremat-2024-11-21_09:15:44-fs:nfs-wip-vshankar-testing-20241118.055430-debug-distro-default-smithi/

@vshankar 
Have you see this error before ? It doesn't seem to be related to the fix ?

Reply:
> Have you see this error before ? It doesn't seem to be related to the fix ?

Looks unrelated, but I'll have a look at the logs in detail.